backend/src/test/scala/cromwell/backend/wdl/ReadLikeFunctionTest.scala

```
@@ -0,0 +1,101 @@
+
+    new StandardExpressionFunctions(dp) {
+      override val readLinesLimit = _readLinesLimit
+      override val readIntLimit = _readIntLimit
+      override val readFloatLimit = _readFloatLimit
+      override val readStringLimit = _readStringLimit
+      override val readJsonLimit = _readJsonLimit
+      override val readTsvLimit = _readTsvLimit
+      override val readMapLimit = _readMapLimit
+      override val readObjectLimit = _readObjectLimit
+    }
+  }
+
+  def testOverUnder(command: String, n: Int, f: ReadLikeFunctions => (Seq[Try[WdlValue]] =>Try[WdlValue])) = {
+
+    def testInner(n: Int, f: ReadLikeFunctions => (Seq[Try[WdlValue]] =>Try[WdlValue]),g: Try[WdlValue] => Unit) = {
+
+      def createTempFileOfSize(n: Int): Path = {
+
+        val tempDir = Files.createTempDir
+        val fn = tempDir.toString + "/whatever"
+        val jPath = Paths.get(fn)
+        val start = Stream[Task, Byte](1).repeat.take(n.toLong)
+        val end = fs2.io.file.writeAll[Task](jPath, Seq(CREATE_NEW, WRITE) )
+        (start to end).run.unsafeRunSync 
+        //jPath is now a file of n bytes, we can return it
+        jPath
+      }
+
+      val file = createTempFileOfSize(n)
+      val params = Seq(Try(WdlString(file.toString)))
+      //apply the test "g"
+      f(rlf)(params) match {
+        case t => g(t)
+      }
+      delete(file)
+    }
+
+    def testOver(n: Int, f: ReadLikeFunctions => (Seq[Try[WdlValue]] =>Try[WdlValue])) = {
+      testInner(n + 1, f, {
+        case Failure(s: FileSizeTooBig) => //success
+        case t => throw new RuntimeException(s"should not have eaten this file that is too big! msg: $t")
+      })
+    }
+
+    def testUnder(n: Int, f: ReadLikeFunctions => (Seq[Try[WdlValue]] =>Try[WdlValue])) = {
+      testInner(n - 1, f, {
+        case Success(_) => 
+        case Failure(s:FileSizeTooBig) => throw s
+        case Failure(t) => //we're not testing parsing
```

I'm sure it's working as intended! :)

But, if it wasn't (or is broken in the future), that error might be swallowed by the "all `Failure(_)`s are ok" policy.